### PR TITLE
Applied Gradle Plugins Declaratively for Most Remaining Plugin Example Apps (Part 3)

### PR DESCRIPTION
**Pull request by jesswrd**
Updated applying gradle plugins from usage of imperative apply to usage of declarative blocks {} apply for 25 remaining plugin example apps. Intending on updating all android example apps under packages. 

Here are previous bulk migrations from imperative apply to declarative apply:
https://github.com/flutter/packages/pull/7968
https://github.com/flutter/packages/pull/8019

Previously migrated applying path_provider plugin example app from imperatively to declaratively in this PR [here](https://github.com/flutter/packages/pull/7822). Tests for changes in the linked PR above.

More information on Flutter Gradle Pluggin Apply [here](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply)

All instances of `minSdkVersion` for example plugin apps use `flutter.minSdkVersion` instead of the hard-coded version https://github.com/flutter/packages/pull/8035.

Partially addresses [#152656](https://github.com/flutter/flutter/issues/152656)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/camera/camera/example/android/app/build.gradle [hunk 1]
```diff
@@ -21,18 +22,15 @@ if (flutterVersionName == null) {
...
 android {
     namespace 'io.flutter.plugins.cameraexample'
     compileSdk flutter.compileSdkVersion
 
 
     defaultConfig {
         applicationId "io.flutter.plugins.cameraexample"
-        minSdkVersion flutter.minSdkVersion
-        targetSdkVersion 28
+        minSdkVersion 21
```
Comment (jmagman) [inline packages/camera/camera/example/android/app/build.gradle:32]: Isn't this undoing what @stuartmorgan just did in https://github.com/flutter/packages/pull/8035?

Comment (jesswrd) [reply packages/camera/camera/example/android/app/build.gradle]: yup nice catch. done.

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by reidbaker** [comment]
@jesswrd can you also update the pr description to include any other changes in this pr. Like changes to min or target sdk.

**Review by reidbaker — APPROVED** [review]

**Review by stuartmorgan — APPROVED** [review]
LGTM

As a follow-up PR, it would probably be good to change the repo tooling's gradle check to ensure that examples are using this plugin style to ensure we don't accidentally add a new example using the old style.

**Review by tarrinneal — APPROVED** [review]

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

**Comment by jesswrd** [comment]
@camsim99 CI failure is unrelated to current PR. @stuartmorgan Can I land on red?

Here are potentially related issues:

https://github.com/flutter/flutter/issues/154682
https://github.com/flutter/flutter/issues/157556

**Comment by matanlurey** [comment]
> CI failure is unrelated to current PR. Can I land on red?

Drive-by answer: In general I would not ignore CI failures and land on red. In this particular case their might not be any consequences, but it's easy to accidentally make a bad situation worse, often in hidden or complex ways.

If there is a test that is not stable enough (i.e. a re-run doesn't turn it green), as potentially mentioned in the above issues, I'd disable the test for now, and merge this PR with a truly green CI status.

(There are exceptional cases where this guidance might not hold - like a security vulnerability or severe regression where the cost of potentially breaking a test is lower than _not_ landing code, but I wouldn't find updating example apps to be one).

I am also happy to review a PR temporarily disabling a flaky test.

**Comment by jesswrd** [comment]
Gotcha, I'll send that PR temporarily disabling the flaky tests. Thanks!

**Comment by jesswrd** [comment]
@matanlurey Gotcha, I'll send that PR temporarily disabling the flaky tests. Thanks!

**Comment by stuartmorgan** [comment]
> @camsim99 CI failure is unrelated to current PR.

Is it?

> Here are potentially related issues:
> [...] [flutter/flutter#157556](https://github.com/flutter/flutter/issues/157556)

This PR changes the target SDK to a value >=31, and you are referencing an issue saying the tests break if the target SDK is >=31. So isn't this PR breaking the integration tests?

> @stuartmorgan Can I land on red?

No, we don't land on red unless it's to fix the tree.

**Comment by jesswrd** [comment]
> This PR changes the target SDK to a value >=31, and you are referencing an issue saying the tests break if the target SDK is >=31. So isn't this PR breaking the integration tests?

I have to bump the target sdk version to at least 33 to avoid an error. Will skip related working tests in this PR and add a note in the PR description.

**Pull request merged**